sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -32,13 +32,9 @@
      * If these environment variables are not available, then this will use the Shared MSAL
      * token cache.
      *
-     * @param identityClientOptions the options to configure the IdentityClient
+     * @param tokenCredentials the list of credentials to execute for authentication.
      */
-    DefaultAzureCredential(IdentityClientOptions identityClientOptions) {
-        super(new ArrayDeque<>(Arrays.asList(new EnvironmentCredential(identityClientOptions),
-            new ManagedIdentityCredential(null, identityClientOptions),
-            new SharedTokenCacheCredential(null, null, "04b07795-8ddb-461a-bbee-02f9e1bf7b46",
-                identityClientOptions),
-            new AzureCliCredential(identityClientOptions))));
+    DefaultAzureCredential(List<TokenCredential> tokenCredentials) {
```

Comment:
Is there a reason to build a List in the builder and convert to ArrayDeque here?